com.sap.cloud.lm.sl.cf.web/src/main/java/com/sap/cloud/lm/sl/cf/web/configuration/SecurityConfiguration.java

```diff
@@ -0,0 +1,79 @@
+package com.sap.cloud.lm.sl.cf.web.configuration;
+
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+
+import javax.inject.Inject;
+import javax.servlet.http.HttpServletRequest;
+import javax.sql.DataSource;
+import javax.ws.rs.core.HttpHeaders;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.access.vote.AuthenticatedVoter;
+import org.springframework.security.access.vote.UnanimousBased;
+import org.springframework.security.oauth2.provider.error.OAuth2AuthenticationEntryPoint;
+import org.springframework.security.oauth2.provider.token.store.JdbcTokenStore;
+import org.springframework.security.web.AuthenticationEntryPoint;
+import org.springframework.security.web.access.expression.WebExpressionVoter;
+import org.springframework.security.web.authentication.DelegatingAuthenticationEntryPoint;
+import org.springframework.security.web.authentication.www.BasicAuthenticationEntryPoint;
+import org.springframework.security.web.util.matcher.RequestMatcher;
+
+import com.sap.cloud.lm.sl.cf.web.security.TokenStoreFactory;
+
+@Configuration
+public class SecurityConfiguration {
+
+    private static final String CF_DEPLOY_SERVICE = "CF Deploy Service";
+
+    @Inject
+    @Bean
+    public DelegatingAuthenticationEntryPoint
+           delegatingAuthenticationEntryPoint(OAuth2AuthenticationEntryPoint oauthAuthenticationEntryPoint,
+                                              BasicAuthenticationEntryPoint basicAuthenticationEntryPoint) {
+        LinkedHashMap<RequestMatcher, AuthenticationEntryPoint> entryPoints = new LinkedHashMap<>();
+        entryPoints.put(this::containsBearerAuthorizationHeader, oauthAuthenticationEntryPoint);
+        DelegatingAuthenticationEntryPoint delegatingAuthenticationEntryPoint = new DelegatingAuthenticationEntryPoint(entryPoints);
+        delegatingAuthenticationEntryPoint.setDefaultEntryPoint(basicAuthenticationEntryPoint);
+        return delegatingAuthenticationEntryPoint;
+    }
+
+    private boolean containsBearerAuthorizationHeader(HttpServletRequest request) {
+        String authorizationHeader = request.getHeader(HttpHeaders.AUTHORIZATION);
+        return authorizationHeader != null ? StringUtils.startsWith(authorizationHeader.toLowerCase(), "bearer") : false;
```

Comment:
org.springframework.util.StringUtils has a method startsWithIgnoreCase. Maybe it will be better? It has built in null check as well.
The apache commons library also has such a method with the null check